pkg/driver/driver.go

```diff
@@ -1259,7 +1279,12 @@ func (d Driver) createNewVolume(ctx context.Context, info *volume.Info, params *
 					"CreateVolume failed for %s: snapshot not found in storage backend", req.GetName())
 			}
 
-			if err := d.Snapshots.VolFromSnap(ctx, snap, info, params, req.GetAccessibilityRequirements()); err != nil {
+			snapParams, err := d.maybeGetSnapshotParameters(ctx, snap)
+			if err != nil {
+				logger.WithError(err).Warn("failed to fetch snapshot parameters, continuing without it")
+			}
+
+			if err := d.Snapshots.VolFromSnap(ctx, snap, info, params, snapParams, req.GetAccessibilityRequirements()); err != nil {
```

Comment:
this feels a bit weird. you might get an `err`, that in this case is not fatal, so you only log it. usually in go if you have `x, err := f()` and you get an `err`, you can not assume anything about `x`. But you then feed `snapParams` into `VolFromSnap`. That just feels wrong, maybe set `snampParams` to `nil`, or to some "defined value" in case of an `err`?